### PR TITLE
Retry balance

### PR DESCRIPTION
**Pull request by enriquesouza**
This will prevent the balance to not load, it should be retried 3 times.

### CODE REVIEW

File: VultisigApp/VultisigApp/Services/Polkadot/PolkadotService.swift
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Services/Polkadot/PolkadotService.swift:26]: Makes sense.

Comment (enriquesouza) [reply VultisigApp/VultisigApp/Services/Polkadot/PolkadotService.swift:26]: I've added the retry when 429 for the GET and Post Methods on utils.

File: VultisigApp/VultisigApp/View Models/VaultDetailViewModel.swift
Comment (johnnyluo) [inline VultisigApp/VultisigApp/View Models/VaultDetailViewModel.swift:44]: since you already retry in the service , like RpcService and PolkadotService , so here you don't need to retry , otherwise it is trying 3*3 times

File: VultisigApp/VultisigApp/Utils/Utils.swift
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Utils/Utils.swift:98]: don't use sub function , you could create a `getRequestWithRetry`

**Review by johnnyluo — CHANGES_REQUESTED** [review]

File: VultisigApp/VultisigApp/Services/BalanceService.swift
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Services/BalanceService.swift:22]: Sure

File: VultisigApp/VultisigApp/Services/Polkadot/PolkadotService.swift
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Services/Polkadot/PolkadotService.swift:44]: changed to ```try await Task.sleep(nanoseconds: retryDelay)```

File: VultisigApp/VultisigApp/View Models/VaultDetailViewModel.swift
Comment (enriquesouza) [reply VultisigApp/VultisigApp/View Models/VaultDetailViewModel.swift:44]: This is also important, because we have Solana, Sui, Thor, Maia, Gaia, Kujira, and others. It will help us when calculating the Super Total. It is also in parallel. So in case we have 10 coins it will start multiple tasks.

Comment (johnnyluo) [reply VultisigApp/VultisigApp/View Models/VaultDetailViewModel.swift:44]: No, what I mean is , if the RPC call to fetch balance is already with retry , then here , we don't need to retry it again , otherwise it is getting multiple retries

**Comment by flypaper0** [comment]
A lot of code that is used here is not exists anymore in refactor PR. I thinks is better to wait until balances refactor will be merged

File: VultisigApp/VultisigApp/Services/BalanceService.swift
Comment (enriquesouza) [reply VultisigApp/VultisigApp/Services/BalanceService.swift]: Done

**Comment by enriquesouza** [comment]
> A lot of code that is used here is not exists anymore in refactor PR. I thinks is better to wait until balances refactor will be merged

ah. oki. true.

**Comment by enriquesouza** [comment]
I will close it for now. If needed we can reuse this.

**Pull request closed without merging**
